packages/datatrak-web/src/features/SurveyResponseModal.tsx

```diff
@@ -0,0 +1,140 @@
+} from '@tupaia/ui-components';
+import { useSurveyResponseWithForm } from '../api/queries';
+import { Button, SurveyTickIcon } from '../components';
+import { displayDate } from '../utils';
+import { SurveyReviewSection } from './Survey/Components';
+import { SurveyContext } from '.';
+
+const Header = styled.div`
+  display: flex;
+  align-items: center;
+  padding: 0.5rem;
+  width: 100%;
+
+  .MuiSvgIcon-root {
+    font-size: 2.5em;
+    margin-right: 0.35em;
+  }
+`;
+
+const Heading = styled(Typography).attrs({
+  variant: 'h2',
+})`
+  font-size: 1.5rem;
+  color: ${({ theme }) => theme.palette.text.primary};
+  font-weight: 600;
+  margin-bottom: 0.2rem;
+  ${({ theme }) => theme.breakpoints.down('sm')} {
+    font-size: 1rem;
+  }
+`;
+
+const SubHeading = styled(Typography)`
+  color: ${({ theme }) => theme.palette.text.secondary};
+  font-weight: 400;
+  font-size: 1rem;
+  ${({ theme }) => theme.breakpoints.down('sm')} {
+    font-size: 0.875rem;
+  }
+`;
+
+const Loader = styled(SpinningLoader)`
+  width: 25rem;
+  max-width: 100%;
+`;
+
+const getSubHeadingText = surveyResponse => {
+  if (!surveyResponse) {
+    return null;
+  }
+  const date = displayDate(surveyResponse.dataTime);
+  const country = surveyResponse?.countryName;
+  const entity = surveyResponse?.entityName;
+  const location = country === entity ? country : `${entity} | ${country}`;
+  return `${location} ${date}`;
+};
+
+// Needs to be wrapped in a context provider to provide the form data to the form
+const SurveyResponseModalContent = ({ onClose }: { onClose: () => void }) => {
+  const [urlSearchParams] = useSearchParams();
+
+  const surveyResponseId = urlSearchParams.get('responseId');
+
+  const {
+    data: surveyResponse,
+    isLoading,
+    isFetched,
+    error,
+  } = useSurveyResponseWithForm(surveyResponseId);
+
+  const subHeading = getSubHeadingText(surveyResponse);
+
+  const showLoading = isLoading || !isFetched;
+
+  return (
+    <>
+      <ModalHeader onClose={onClose} title={error ? 'Error loading survey response' : ''}>
+        {!showLoading && !error && (
+          <Header>
+            <SurveyTickIcon />
+            <div>
+              <Heading>{surveyResponse?.surveyName}</Heading>
+              <SubHeading>{subHeading}</SubHeading>
+            </div>
+          </Header>
+        )}
+      </ModalHeader>
+      <ModalContentProvider errorMessage={(error as Error)?.message}>
+        {showLoading && <Loader />}
+        {!showLoading && !error && <SurveyReviewSection />}
+      </ModalContentProvider>
+      <ModalFooter>
+        <Button to="/">Close</Button>
+      </ModalFooter>
+    </>
+  );
+};
+
+export const SurveyResponseModal = () => {
+  const formContext = useForm();
+  const [urlSearchParams, setUrlSearchParams] = useSearchParams();
+
+  const surveyResponseId = urlSearchParams.get('responseId');
+  const surveyCode = urlSearchParams.get('surveyCode');
+  const countryCode = urlSearchParams.get('countryCode');
```

Comment:
```suggestion
```
Since the surveyCode and the countryCode can be derived from the surveyResponseId, I don't think that we should include them in the url. I think that it is good practise to have the least possible amount of url query string params possible and since we are already doing a network request before loading the form, there should be a good place to get that data